com/williamfiset/algorithms/datastructures/linkedlist/DoublyLinkedList.java

```diff
@@ -77,6 +77,33 @@ public void addFirst(T elem) {
     size++;
   }
 
+  // Add an element at a specified index
+  public void addAt(int index, T data) throws Exception {
+    if (index < 0) {
+      throw new Exception("Illegal Index");
+    }
+    if (index == 0) {
+      addFirst(data);
+      return;
+    }
+
+    if (index == size) {
+      addLast(data);
+      return;
+    }
+
+    Node<T> temp = head;
+    for (int i = 0; i < index - 1; i++) {
+      temp = temp.next;
+    }
+    Node<T> newNode = new Node(data, null, null);
```

Comment:
I think you can simplify this by passing in temp and temp.next into the constructor for the new node:

Node<T> newNode = new Node(data, temp, temp.next);

I believe we also need to set temp.next.prev = newNode so that we can keep the doubly linked structure.

Node<T> newNode = new Node(data, temp, temp.next);
temp.next.prev = newNode;
temp.next = newNode;